src/irc.cc

```diff
@@ -69,6 +70,22 @@ void addmessage(struct ctx *ctxptr, char *nick, const char *text) {
 		}
 		lastmsg->next = newmsg;
 	}
+	if(ctxptr->max_msg_lines>0) {
+		int msgcnt = 0;
+		newmsg = ctxptr->messages;
+		while(newmsg) {
```

Comment:
ok, I can assume some upper bound for max_msg_lines and msg_count - this should be reasonable. However, the loop in line 68 to 70 will last indefinitely if there is no null pointer, anyway.